src/NumericalAlgorithms/Spectral/SwshTransform.hpp

```diff
@@ -155,248 +412,248 @@ static const size_t max_libsharp_transforms = 100;
-// transforms using `conjugate_views` to restore the input data to its original
-// state.
-template <ComplexRepresentation Representation>
-void append_libsharp_collocation_pointers(
-    gsl::not_null<std::vector<double*>*> collocation_data,
-    gsl::not_null<std::vector<ComplexDataView<Representation>>*>
-        collocation_views,
-    gsl::not_null<ComplexDataVector*> vector, size_t l_max,
-    bool positive_spin) noexcept;
-
-// Perform a conjugation on a vector of `ComplexDataView`s if `Spin` < 0. This
-// is used for undoing the conjugation described in the code comment for
-// `append_libsharp_collocation_pointers`
-template <int Spin, ComplexRepresentation Representation>
-SPECTRE_ALWAYS_INLINE void conjugate_views(
-    gsl::not_null<std::vector<ComplexDataView<Representation>>*>
-        collocation_views) noexcept {
-  if (Spin < 0) {
-    for (auto& view : *collocation_views) {
-      view.conjugate();
-    }
+/*!
+ * \ingroup SwshGroup
+ * \brief A \ref DataBoxGroup mutate-compatible computational struct for
+ * performing several spin-weighted inverse spherical harmonic transforms.
+ * Internally dispatches to libsharp.
+ *
+ * \details
+ * Template Parameters:
+ * - `Representation`: The element of the `ComplexRepresentation` enum which
+ *   parameterizes how the data passed to libsharp will be represented. Two
+ *   options are available:
+ *  - `ComplexRepresentation:Interleaved`: indicates that the real and imaginary
+ *    parts of the collocation values will be passed to libsharp as pointers
+ *    into existing complex data structures, minimizing copies, but maintaining
+ *    a stride of 2 for 'adjacent' real or imaginary values.
+ *  - `ComplexRepresentation::RealsThenImags`: indicates that the real and
+ *    imaginary parts of the collocation values will be passed to libsharp as
+ *    separate contiguous blocks. At current, this introduces both allocations
+ *    and copies.  **optimization note** In the future most of the allocations
+ *    can be aggregated to calling code, which would pass in buffers by
+ *    `not_null` pointers.
+ *
+ *  For performance-sensitive code, both options should be tested, as each
+ *  strategy has trade-offs.
+ * - `TagList`: A `tmpl::list` of Tags to be inverse transformed. The tags must
+ * represent the nodal data being transformed to.
+ *
+ * \see `SwshTransform` for mathematical notes regarding the libsharp modal
+ * representation taken as input for this computational struct.
+ */
+template <typename TagList, ComplexRepresentation Representation =
+                                ComplexRepresentation::Interleaved>
+struct InverseSwshTransform;
+
+/// \cond
+template <typename... TransformTags, ComplexRepresentation Representation>
+struct InverseSwshTransform<tmpl::list<TransformTags...>, Representation> {
+  static constexpr int spin =
+      db::item_type<tmpl::front<tmpl::list<TransformTags...>>>::type::spin;
+
+  static_assert(
```

Comment:
if you decide to change the `SwishTransform` check at all, please change this one as well :)